api/file.go

```diff
@@ -143,25 +146,58 @@ func fireAndForgetHandleImages(filenames []string, fileData [][]byte, teamId, ch
 					return
 				}
 
-				// Decode image config
-				imgConfig, _, err := image.DecodeConfig(bytes.NewReader(fileData[i]))
-				if err != nil {
-					l4g.Error("Unable to decode image config channelId=%v userId=%v filename=%v err=%v", channelId, userId, filename, err)
-					return
+				width := img.Bounds().Dx()
+				height := img.Bounds().Dy()
+
+				// Get the image's orientation and ignore any errors since not all images will have orientation data
+				orientation, _ := getImageOrientation(fileData[i])
+
+				// Create a temporary image that will be manipulated and then used to make the thumbnail and preview image
+				var temp *image.RGBA
+				if orientation >= 1 && orientation <= 4 {
```

Comment:
What do `1` and `4` mean here? Is possible to use constants or an enumerated type to make it more clear?